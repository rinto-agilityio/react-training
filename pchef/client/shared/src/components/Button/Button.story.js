// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Helper
import { wInfo } from '../../../.storybook/utils';

// Component
import Button from './index'

storiesOf('Button', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Button
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Success', () => (
    <Button
      type="success"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Danger', () => (
    <Button
      type="danger"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Warning', () => (
    <Button
      type="warning"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))

  .add('Info', () => (
    <Button
      type="info"
      title="Click me"
      onPress={action('clicked')}
      buttonStyle={{
        width: 100,
      }}
    />
  ))
