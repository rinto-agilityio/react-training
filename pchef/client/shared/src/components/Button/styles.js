// Lib
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../themes'

// Create style for button
const styles = StyleSheet.create({
  button: {
    ...METRICS.flexCenter,
    padding: METRICS.padding.sm,
  },
  title: {
    color: COLORS.white,
    textTransform: 'uppercase',
    fontWeight: FONTS.fontWeight.large,
  },
  default: {
    backgroundColor: COLORS.baseBlue,
  },
  success: {
    backgroundColor: COLORS.green,
  },
  danger: {
    backgroundColor: COLORS.red,
  },
  warning: {
    backgroundColor: COLORS.yellow,
  },
  info: {
    backgroundColor: COLORS.darkWhite,
  },
})

export default styles
