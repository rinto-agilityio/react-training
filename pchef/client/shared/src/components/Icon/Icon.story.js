// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Themes
import { COLORS } from '../../themes'

// Components
import Icon from '.'

// Helper
import { wInfo } from '../../../.storybook/utils';

storiesOf('Icon', module)
  .addDecorator(wInfo())
  .add('Default', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.icon.default}
      name="add-a-photo"
    />
  ))
  .add('Disabled', () => (
    <Icon
      onPress={action('clicked')}
      color={COLORS.icon.default}
      name="add-a-photo"
      disabled
    />
  ))
