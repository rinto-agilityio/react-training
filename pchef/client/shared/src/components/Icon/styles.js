// Lib
import { StyleSheet } from 'react-native'

// Themes
import { METRICS } from '../../themes'

// Create style for Icon
const styles = StyleSheet.create({
  wrapIcon: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  label: {
    marginLeft: METRICS.mediumMargin,
    textAlign: 'center',
  },
})

export default styles
