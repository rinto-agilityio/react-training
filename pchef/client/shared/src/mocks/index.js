// mock categories
export const category = {
  name: 'Heathy',
  imgUrl:
    'https://images.pexels.com/photos/461198/pexels-photo-461198.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500',
}
