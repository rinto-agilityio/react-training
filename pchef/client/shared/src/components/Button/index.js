// libs
import React from 'react'
import { Button as ButtonComponent, Text } from 'react-native-paper'

// Styles
import styles from './styles'
import { COLORS } from '../../themes/index'

type Props = {
  onPress?: () => void,
  buttonStyle?: {} | Array<{}>,
  title: string,
  titleStyle?: {} | Array<{}>,
  disabled?: boolean,
  mode?: string,
  contentStyle?: {} | Array<{}>,
  color?: string,
  loading?: boolean,
  type?: string,
}

const Button = ({
  onPress,
  buttonStyle,
  title,
  mode,
  type,
  disabled,
  contentStyle,
  color,
  loading,
  titleStyle,
}: Props) => (
  <ButtonComponent
    color={color}
    loading={loading}
    onPress={onPress}
    mode={mode}
    disabled={disabled}
    style={[styles.button, styles[type], buttonStyle]}
    contentStyle={[styles.content, contentStyle]}
  >
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </ButtonComponent>
)

Button.defaultProps = {
  onPress: () => {},
  buttonStyle: {},
  titleStyle: {},
  disabled: false,
  mode: 'contained',
  contentStyle: {},
  color: COLORS.baseBlue,
  loading: false,
  type: 'default',
}

export default Button
