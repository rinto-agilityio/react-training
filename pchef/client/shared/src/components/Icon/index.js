// Libs
import React from 'react'
import { View, Text } from 'react-native'
import { IconButton } from 'react-native-paper'

// Styles
import styles from './styles'

// Themes
import { COLORS, FONTS } from '../../themes'

type Props = {
  label?: string | number,
  onPress?: () => void | Promise<void>,
  customStyle?: {} | Array<{}>,
  color?: string,
  size?: number,
  name: string,
  disabled?: boolean,
  wrapperIconStyle?: Object,
  customIconStyles?: {},
};

const Icon = ({
  label,
  onPress,
  customStyle,
  color,
  size,
  name,
  wrapperIconStyle,
  disabled,
  customIconStyles,
}: Props) => (
  <View style={[
    styles.wrapIcon,
    wrapperIconStyle,
  ]}
  >
    <IconButton
      icon={name}
      color={color}
      size={size}
      onPress={onPress}
      disabled={disabled}
      style={[styles.iconStyles, customIconStyles]}
    />
    {label ? <Text style={[styles.label, customStyle]}>{label}</Text> : null}
  </View>
)

Icon.defaultProps = {
  label: '',
  onPress: () => {},
  customStyle: {},
  size: FONTS.fontSize.medium,
  color: COLORS.black,
  wrapperIconStyle: {},
  disabled: false,
  customIconStyles: {},
};

export default Icon
