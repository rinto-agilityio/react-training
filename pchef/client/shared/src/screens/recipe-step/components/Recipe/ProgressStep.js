// Libs
import React from 'react'

// Styles
import styles from './styles'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Components
import Button from '../../../../components/Button'
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'

type Props = {
  steps: Array<{
    step: number
  }>,
  size: string,
  step: number,
  onPressStep?: () => void,
  onPressSelectStep?: () => void,
  customStepBtn?: {},
}

const chevronIcon = (
  name: string,
  size: string,
  disabled: boolean,
  onPress: () => void
) => (
  <Icon
    name={name}
    size={FONTS.fontSize[`${size}`]}
    color={disabled ? COLORS.baseGray : COLORS.baseBlue}
    onPress={onPress}
    disabled={disabled}
    customIconStyles={{
      margin: 0,
    }}
  />
)

const Progress = ({
  steps = [],
  size = 'large',
  step,
  onPressStep,
  customStepBtn,
  onPressSelectStep,
}: Props) => {
  // Disabled prev/next icon if current step is first/last step
  const disablePrevIcon = step === 1
  const disabledNextIcon = steps.length === step

  return (
    <Wrapper direction="row">
      {chevronIcon('chevron-left', size, disablePrevIcon, onPressSelectStep)}
      <Wrapper direction="row">
        {steps.map(item => {
          const status = step === item.step ? 'active' : 'inactive'

          return (
            <Button
              key={item.step}
              onPress={onPressStep}
              buttonStyle={[
                styles.progressStep,
                styles[`${status}Step`],
                customStepBtn,
              ]}
              contentStyle={styles[`${size}Step`]}
              title=""
            />
          )
        })}
      </Wrapper>
      {chevronIcon('chevron-right', size, disabledNextIcon, onPressSelectStep)}
    </Wrapper>
  )
}

Progress.defaultProps = {
  onPressStep: () => {},
  customStepBtn: {},
  onPressSelectStep: () => {},
}

export default Progress
