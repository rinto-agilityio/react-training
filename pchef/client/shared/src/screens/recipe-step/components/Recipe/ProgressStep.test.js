// Components
import ProgressStep from './ProgressStep'
import Button from '../../../../components/Button'
import Icon from '../../../../components/Icon'

// Mocks
import { recipes } from '../../../../mocks'

const stepProps = {
  steps: recipes[0].steps,
  step: 1,
  onPressStep: jest.fn(),
  onPressSelectStep: jest.fn(),
}

describe('Recipe by step', () => {
  let progressStep

  beforeEach(() => {
    progressStep = shallow(<ProgressStep {...stepProps} />)
  })

  it('Renders correctly progress steps commponent', () => {
    const progressStepComponent = renderer.create(
      <ProgressStep {...stepProps} />
    ).toJSON
    expect(progressStepComponent).toMatchSnapshot()
  })

  it('Render progress step component with steps array', () => {
    const props = {
      ...stepProps,
      size: 'medium',
    }
    const progressStepComponent = renderer
      .create(<ProgressStep {...props} />)
      .toJSON()
    expect(progressStepComponent).toMatchSnapshot()
  })

  it('Should call onPressStep when press on Button of ProgressStep', () => {
    progressStep.setProps({ size: 'medium' })
    progressStep
      .find(Button)
      .at(0)
      .props()
      .onPress()
    expect(stepProps.onPressStep).toHaveBeenCalled()
  })

  it('Should call onPressSelectStep when press on prev icon', () => {
    progressStep
      .find(Icon)
      .at(0)
      .props()
      .onPress()
    expect(stepProps.onPressSelectStep).toHaveBeenCalled()
  })

  it('Should call onPressSelectStep when press on next icon', () => {
    progressStep
      .find(Icon)
      .at(1)
      .props()
      .onPress()
    expect(stepProps.onPressSelectStep).toHaveBeenCalled()
  })
})
