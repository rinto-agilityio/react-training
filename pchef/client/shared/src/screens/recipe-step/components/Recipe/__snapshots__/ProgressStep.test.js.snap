// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe by step Render progress step component with steps array 1`] = `
<div
  className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-justifyContent-a2tzq0"
  style={
    Object {
      "WebkitBoxFlex": 0,
      "WebkitFlexGrow": 0,
      "flexGrow": 0,
      "msFlexPositive": 0,
    }
  }
>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
  >
    <div
      aria-disabled={true}
      className="css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-height-1r8g8re r-justifyContent-1777fci r-opacity-qzm50q r-overflow-bztko3 r-userSelect-lrvibr r-width-1acpoxo"
      disabled={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        Object {
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
        }
      }
    >
      <span
        className="css-hitSlop-mjp8i1"
        style={
          Object {
            "bottom": "-6px",
            "left": "-6px",
            "right": "-6px",
            "top": "-6px",
          }
        }
      />
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
          dir="auto"
          focusable={false}
          role="img"
          style={
            Object {
              "WebkitTransform": "scaleX(1)",
              "color": "rgba(197,197,197,1.00)",
              "fontFamily": "MaterialIcons",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "transform": "scaleX(1)",
            }
          }
        >
          î—‹
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-display-6koalj r-flexDirection-18u37iz r-flexWrap-1w6e6rj r-justifyContent-a2tzq0"
    style={
      Object {
        "WebkitBoxFlex": 0,
        "WebkitFlexGrow": 0,
        "flexGrow": 0,
        "msFlexPositive": 0,
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(66,165,245,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(197,197,197,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(197,197,197,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(197,197,197,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(197,197,197,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
      style={
        Object {
          "backgroundColor": "rgba(197,197,197,1.00)",
          "borderBottomColor": "rgba(0,0,0,0.00)",
          "borderBottomLeftRadius": "4px",
          "borderBottomRightRadius": "4px",
          "borderBottomStyle": "solid",
          "borderBottomWidth": "0px",
          "borderLeftColor": "rgba(0,0,0,0.00)",
          "borderLeftStyle": "solid",
          "borderLeftWidth": "0px",
          "borderRightColor": "rgba(0,0,0,0.00)",
          "borderRightStyle": "solid",
          "borderRightWidth": "0px",
          "borderTopColor": "rgba(0,0,0,0.00)",
          "borderTopLeftRadius": "4px",
          "borderTopRightRadius": "4px",
          "borderTopStyle": "solid",
          "borderTopWidth": "0px",
          "boxShadow": "0px 0.75px 1.5px rgba(0,0,0,0.24)",
          "marginRight": "3px",
          "minWidth": "0px",
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="css-cursor-18t94o4 css-view-1dbjc4n r-backgroundColor-1niwhzg r-cursor-1loqt21 r-touchAction-1otgn73 r-userSelect-lrvibr"
        data-focusable={true}
        disabled={false}
        onKeyDown={[Function]}
        onKeyPress={[Function]}
        onKeyUp={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        style={
          Object {
            "borderBottomLeftRadius": "4px",
            "borderBottomRightRadius": "4px",
            "borderTopLeftRadius": "4px",
            "borderTopRightRadius": "4px",
            "overflowX": "hidden",
            "overflowY": "hidden",
          }
        }
        tabIndex="0"
      >
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-height-4v7adb r-justifyContent-1777fci r-width-199wky7"
        >
          <div
            className="css-text-901oao css-textOneLine-bfa6kz r-letterSpacing-ud0q2t r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr r-textAlign-q4m81j"
            dir="auto"
            style={
              Object {
                "color": "rgba(0,0,0,1.00)",
                "direction": "ltr",
                "fontFamily": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
              }
            }
          >
            <span
              className="css-text-901oao css-textHasAncestor-16my406 r-color-jwli3a r-fontWeight-1kfrs79 r-textTransform-tsynxw"
              dir="auto"
              style={
                Object {
                  "direction": "ltr",
                  "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                  "textAlign": "left",
                }
              }
            >
              
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
  >
    <div
      className="css-cursor-18t94o4 css-view-1dbjc4n r-alignItems-1awozwy r-backgroundColor-1niwhzg r-borderRadius-6ncur5 r-cursor-1loqt21 r-height-1r8g8re r-justifyContent-1777fci r-overflow-bztko3 r-touchAction-1otgn73 r-userSelect-lrvibr r-width-1acpoxo"
      data-focusable={true}
      disabled={false}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      role="button"
      style={
        Object {
          "marginBottom": "0px",
          "marginLeft": "0px",
          "marginRight": "0px",
          "marginTop": "0px",
        }
      }
      tabIndex="0"
    >
      <span
        className="css-hitSlop-mjp8i1"
        style={
          Object {
            "bottom": "-6px",
            "left": "-6px",
            "right": "-6px",
            "top": "-6px",
          }
        }
      />
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-text-901oao r-backgroundColor-1niwhzg r-pointerEvents-633pao"
          dir="auto"
          focusable={false}
          role="img"
          style={
            Object {
              "WebkitTransform": "scaleX(1)",
              "color": "rgba(66,165,245,1.00)",
              "fontFamily": "MaterialIcons",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "transform": "scaleX(1)",
            }
          }
        >
          î—Œ
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Recipe by step Renders correctly progress steps commponent 1`] = `[Function]`;
